packages/lambdas/src/document-downloader.ts

```diff
@@ -0,0 +1,312 @@
+import {
+  Document,
+  CommonWell,
+  CommonWellAPI,
+  APIMode,
+  PurposeOfUse,
+} from "@metriport/commonwell-sdk";
+import { getSecret } from "@aws-lambda-powertools/parameters/secrets";
+import * as Sentry from "@sentry/serverless";
+import * as stream from "stream";
+import { PassThrough } from "stream";
+import AWS from "aws-sdk";
+import { capture } from "./shared/capture";
+import { getEnv, getEnvOrFail } from "./shared/env";
+
+// Keep this as early on the file as possible
+capture.init();
+
+// Automatically set by AWS
+const lambdaName = getEnv("AWS_LAMBDA_FUNCTION_NAME");
+// Set by us
+const bucketName = getEnvOrFail("MEDICAL_DOCUMENTS_BUCKET_NAME");
+const cwOrgCertificateSecret = getEnvOrFail("CW_ORG_CERTIFICATE");
+const cwOrgPrivateKeySecret = getEnvOrFail("CW_ORG_PRIVATE_KEY");
+const envType = getEnvOrFail("ENV_TYPE");
+
+const apiMode = envType === "production" ? APIMode.production : APIMode.integration;
+export const OID_PREFIX = "urn:oid:";
+
+const s3client = new AWS.S3({
+  signatureVersion: "v4",
+});
+
+type DocumentWithOriginalIdAndContent = Document & {
+  originalId: string;
+  content: { location: string };
+};
+
+export type ContentMimeType = Pick<Document["content"], "mimeType">;
+
+export type S3Info = {
+  docId: string;
+  fileExists: boolean;
+  fileSize: number | undefined;
+  fileName: string;
+  fileLocation: string;
+};
+
+export const handler = Sentry.AWSLambda.wrapHandler(
+  async (req: {
+    document: DocumentWithOriginalIdAndContent;
+    fileInfo: S3Info;
+    orgName: string;
+    orgOid: string;
+    facilityNPI: string;
+  }) => {
+    const { document, fileInfo, orgName, orgOid, facilityNPI } = req;
+
+    const cwOrgCertificate: string = (await getSecret(cwOrgCertificateSecret)) as string;
+    if (!cwOrgCertificate) {
+      throw new Error(`Config error - CW_ORG_CERTIFICATE doesn't exist`);
+    }
+
+    const cwOrgPrivateKey: string = (await getSecret(cwOrgPrivateKeySecret)) as string;
+    if (!cwOrgPrivateKey) {
+      throw new Error(`Config error - CW_ORG_CERTIFICATE doesn't exist`);
+    }
+
+    const { writeStream, promise } = uploadStream(
+      fileInfo.fileName,
+      fileInfo.fileLocation,
+      document.content.mimeType
+    );
+
+    await downloadDocumentFromCW({
+      orgCertificate: cwOrgCertificate,
+      orgPrivateKey: cwOrgPrivateKey,
+      orgName,
+      orgOid,
+      facilityNPI,
+      location: document.content.location,
+      stream: writeStream,
+    });
+
+    const uploadResult = await promise;
+
+    console.log(`Uploaded ${document.id} to ${uploadResult.Location}`);
+
+    const downloadedDocument = await downloadDocumentFromS3({ fileName: uploadResult.Key });
```

Comment:
I think we could save some time and memory if we use store the file in-memory while we download it.

Originally, the passthrough stream was the best option b/c we didn't need to have the file in memory on the API. But since we need it here, having the passthrough and later load the file in memory seems like we're losing the opportunity to keep it around while downloading.

Examples:
- https://gist.github.com/jtbonhomme/842be13d57ad6e852657d88c681755ec
- https://www.alxolr.com/articles/how-to-fork-a-nodejs-stream-into-many-streams